tools/webhook-apicoverage/resourcetree/resourcetree.go

```diff
@@ -67,4 +69,11 @@ func (r *ResourceTree) BuildResourceTree(t reflect.Type) {
 // UpdateCoverage updates coverage data in the resource tree based on the provided reflect.Value
 func (r *ResourceTree) UpdateCoverage(v reflect.Value) {
 	r.Root.updateCoverage(v)
+}
+
+// GetCoverageData calculates the coverage information for a resource tree by applying provided Node and Field rules.
+func (r *ResourceTree) GetCoverageData(nodeRules NodeRules, fieldRules FieldRules) ([]coveragecalculator.TypeCoverage) {
```

Comment:
why is this not "BuildCoverageData" for the sake of naming consistency?

Reply:
The reason was this was getting exposed to client, hence "Get" made more sense. But not too particular, I have changed it to BuildCoverageData in favor of consistency.